src/main/java/com/google/ehub/servlets/FavoriteItemServlet.java

```diff
@@ -0,0 +1,73 @@
+package com.google.ehub.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.ehub.data.EntertainmentItemDatastore;
+import com.google.ehub.data.FavoriteItemDatastore;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.lang3.math.NumberUtils;
+
+@WebServlet("/favorite-item")
+public class FavoriteItemServlet extends HttpServlet {
+  private static final String FAVORITE_ITEM_ID_PARAMETER_KEY = "favoriteItemId";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (!userService.isUserLoggedIn()) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST, "FavoriteItemServlet: User is not logged in!");
+      return;
+    }
+
+    response.setContentType("application/json");
+    response.getWriter().println(
+        new Gson().toJson(FavoriteItemDatastore.getInstance().queryFavoriteIds(
+            userService.getCurrentUser().getEmail())));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String favoriteItemStringId = request.getParameter(FAVORITE_ITEM_ID_PARAMETER_KEY);
+
+    if (!isPostRequestParameterValid(favoriteItemStringId)) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST,
+          "FavoriteItemServlet: Post request parameter not specified correctly!");
+      return;
+    }
+
+    Long itemId = Long.parseLong(favoriteItemStringId);
+
+    if (!doesItemExist(itemId)) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST,
+          "FavoriteItemServlet: Favorite Item Id doesn't map to a known item!");
+      return;
+    }
+
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (!userService.isUserLoggedIn()) {
```

Comment:
Does userLoggedIn imply user has a profile? A little worried you'll be adding favorites for people who don't actually have profiles which means you won't be able to recommend friend profiles, just the friend email. (not a huge deal, but recommending profiles would be nice, can be done as separate PR)

Reply:
I think the way that it is setup right now it is not really asking if the user has a profile. I am going to ask Oyin if there is a way to know if the user has created the profile and it's logged in at the same time.

Reply:
Sounds good. Another possibility is to force the user to create a profile the first time they login. So all users are guaranteed to have a profile